email-service/cmd/main.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/mail"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/mail/backends"
+	mailCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/mail/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/models"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/notifications"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/email-service/internal/notifications/message"
+	transportCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/cron"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database"
+	dbCfg "github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/database/config"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/settings"
+)
+
+const defaultCronSpec = "0 0 12 * * *"
+
+const emailTimeout = 5 * time.Second
+
+var appConfig appCfg.Config
+
+func InitDatabase() (*database.DB, error) {
+	db, err := database.New(dbCfg.NewFromEnv())
+	if err != nil {
+		return nil, err
+	}
+	if err := db.Migrate(&models.Rate{}, &models.Subscriber{}); err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func InitNotificationsCron(db *database.DB, mailer notifications.EmailClient) *cron.Manager {
+	// Start cron job for notifications
+	spec := appConfig.NotificationCropSpec
+	if spec == "" {
+		spec = defaultCronSpec
+	}
+
+	notifier := notifications.NewMailNotifier(
+		mailer,
+		models.NewRateRepository(db),
+		models.NewSubscriberRepository(db),
+		&message.PlainRate{},
+	)
+
+	notifyF := func() error {
+		ctx, cancel := context.WithTimeout(context.Background(), emailTimeout)
+		defer cancel()
+		return notifier.Notify(ctx)
+	}
+
+	cronManager := cron.NewManager()
+	err := cronManager.AddJob(spec, notifyF)
+	if err != nil {
+		slog.Error("adding job", slog.Any("error", err))
+	}
+	return cronManager
+}
+
+func doWithContext(ctx context.Context, f func() error) error {
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		if err := f(); err != nil {
+			slog.Error("error", slog.Any("error", err))
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-done:
+			return nil
+		}
+	}
+}
+
+func InitRateConsumer(config transportCfg.Config, rateRepo *models.RateRepository) *rate.Client {
+	rateConsumer, err := rate.NewClient(transportCfg.Config{
+		QueueName: appConfig.RateQueueName,
+		BrokerURI: config.BrokerURI,
+	})
+	if err != nil {
+		slog.Error("creating rate client", slog.Any("error", err))
+		return nil
+	}
+	rateConsumer.Subscribe(func( // nolint: errcheck
+		ctx context.Context,
```

Comment:
I don't think this is a good place to write business logic. `main.go` should be as clean/small as possible

see https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pull/9/files#r1669747729

Reply:
also, handle error, please